scripts/wmata_data/distcalc.py

```diff
@@ -0,0 +1,149 @@
+import sys
+import csv
+import requests
+
+#TODO replace these with code4dc keys
+WMATA_HEADERS = {'api_key': ''}
+MAPBOX_PARAMS = {'access_token': ''}
+
+###### Converts meters to miles
+def getMiles(i):
+    return i*0.000621371192
+
+###### Converst miles to meters
+def getMeters(i):
+    return i*1609.344
+
+###### Gets walking distance between two locations in meters
+def getWalkingDistance(srcLat, srcLon, destLat, destLon):
+    distReqCoords = srcLon + ',' + srcLat + ';' + destLon + ',' + destLat;
+
+    #according to documentation, this doesn't work in Python SDK so switched to using REST API
+    walkDistResponse = requests.get("https://api.mapbox.com/directions/v5/mapbox/walking/"+distReqCoords,
+                                params=MAPBOX_PARAMS)
+    return walkDistResponse.json()['routes'][0]['legs'][0]['distance']
+
+
+###### Finds all rail stations within {radiusinmeters} from the given project. Writes to given CSV file.
+def findRailStations(railStations,project,radiusinmeters,distCsvWriter):
+    lat = project['Proj_lat']
+    lon = project['Proj_lon']
+
+    for station in railStations:
+        walkDist = getWalkingDistance(lat, lon, str(station['Lat']), str(station['Lon']))
+        if walkDist <= radiusinmeters:
+            distCsvWriter.writerow((project['Nlihc_id'],'rail',station['Code'],"{0:.2f}".format(getMiles(walkDist))))
+
+
+###### Finds all bus stations within {radiusinmeters} from the given project. Writes to given CSV file.
+def findBusStations(project, radiusinmeters, distCsvWriter):
+    lat = project['Proj_lat']
+    lon = project['Proj_lon']
+
+    params = {'Lat': lat,
+              'Lon' : lon,
+              'Radius': str(radiusinmeters)}
+    response = requests.get('https://api.wmata.com/Bus.svc/json/jStops',params=params, headers=WMATA_HEADERS)
+    data = response.json()
+
+    for stop in data['Stops']:
+        walkDist = getWalkingDistance(lat, lon, str(stop['Lat']), str(stop['Lon']))
+
+        if walkDist <= radiusinmeters: #within 0.5 miles walking
+            distCsvWriter.writerow((project['Nlihc_id'],'bus',stop['StopID'],"{0:.2f}".format(getMiles(walkDist))))
+
+
+####### Writes rail station metadata to given CSV writer
+def writeRailInfo(railStations, infoCsvWriter):
+
+    print("Writing RAIL INFO");
+
+    for station in railStations:
+        #delimit list of lines with colon
+        lines = station['LineCode1'] #all stations have at least one line
+
+        #there is probably a more elegant way to write this...
+        if station['LineCode2'] != None:
+            lines = '{}:{}'.format(lines, station['LineCode2'])
```

Comment:
For this context, better way is `lines += station['LineCode2']` or maybe you might need to use `lines += str(station['LineCode2'])` depending whether the parsed json needs explicit string coercion. `a+=b` is equivalent to `a = a + b`